src/main/java/io/reactivex/observers/TestObserver.java

```diff
@@ -533,6 +533,43 @@ private AssertionError fail(String message) {
         return this;
     }
 
+    /**
+     * Asserts that this TestObserver received exactly one onNext value for which
+     * the provided predicate returns true.
+     * @param valuePredicate
+     *            the predicate that receives the onNext value
+     *            and should return true for the expected value.
+     * @return this
+     */
+    public final TestObserver<T> assertValue(Predicate<T> valuePredicate) {
+        int s = values.size();
+        if (s == 0) {
+            throw fail("No values");
+        }
+
+        boolean found = false;
+
+        for (T value : values) {
+            try {
+                if (valuePredicate.test(value)) {
```

Comment:
Because the first error may come from the lack of onSubscribe call or some other unexpected exception. Knowing that the exception is there but not the only one is different from expecting exactly 1 element but then getting n.